ompi/request/req_test.c

```diff
@@ -204,21 +204,23 @@ int ompi_request_default_test_all(
     ompi_request_t **rptr;
     size_t num_completed = 0;
     ompi_request_t *request;
+    int do_it_once = 0;
 
     opal_atomic_mb();
     rptr = requests;
     for (i = 0; i < count; i++, rptr++) {
         request = *rptr;
-
-        if( request->req_state == OMPI_REQUEST_INACTIVE ||
-            REQUEST_COMPLETE(request) ) {
+#if OPAL_ENABLE_PROGRESS_THREADS == 0
+recheck_request_status:
+#endif
```

Comment:
I think you need to. reset num_completed to 0 here as well

Reply:
No need to reset, the code stays in the same loop. The PR text describes more clearly the mechanism, the loop is only done once, the first time it encounters an active, non-completed request calls opal_progress, and then if anything new happened it continues the loop starting from the current element.